pkg/cli/start_jemalloc.go

```diff
@@ -19,7 +19,8 @@
 package cli
 
 // #cgo CPPFLAGS: -DJEMALLOC_NO_DEMANGLE
-// #cgo LDFLAGS: -ljemalloc
+// #cgo !darwin LDFLAGS: -ljemalloc
+// #cgo darwin LDFLAGS: -ljemalloc -u_je_zone_register
```

Comment:
> Hm, this behaviour is specific to macOS, which we don't run CI on. How would we test it?

Developers run tests on macOS frequently. A test would have caught this very quickly. Also, seems possible for this to break on another platform as well.